values/option.go

```diff
@@ -0,0 +1,10 @@
+package values
+
+type Option struct {
```

Comment:
Can you add a comment explaining how this works? Why does a value need to know if its an option?

Reply:
Yes. I'll give the explanation above.